lib/http/node.js

```diff
@@ -123,8 +123,14 @@ function getProxyAgent(proxyOptions, parsedUrl, destination, mock) {
       return mockAgent;
     }
   }
-
-  const agentId = createAgentId(agentOptions.protocol, agentOptions.hostname, agentOptions.keepAlive);
+  let destHost;
+  try {
+    destHost = new URL(destination).hostname;
+  } catch (err) {
+    destHost = destination;
+  }
+  Logger.getInstance().debug(`Destination is ${destHost}`);
+  const agentId = createAgentId(agentOptions.protocol, agentOptions.hostname, destHost, agentOptions.keepAlive);
```

Comment:
it means that we will always have hostname in the key but it only affects the cases when proxy is used - do we want to have a property in driver to control it?

Reply:
i personally would vote for always having the same structure of the key and not have a separate setting. also would help with not needing to rewrite the applications built on top of snowflake-sdk if we would happen to need to expose it. if no such setting, driver always uses the same cache key structure, then this change can help transparently already

Reply:
https://github.com/snowflakedb/snowflake-connector-nodejs/pull/946/files#diff-e251ed9eb74ad284dfcd0fcc9a3d2e34d6b18d4b5cba06f21f040beeb0573013R10

Reply:
Please let me know if I need to add an option for this. I agree with @sfc-gh-dszmolka's opinion.

Reply:
@sfc-gh-dprzybysz since this is _the_ core of this whole change; to always add a destination to agent cache key, for being able to utilize separate settings for separate destination, do you agree to proceed like this?